webpack/components/extensions/HostDetails/Tabs/ModuleStreamsTab/ModuleStreamsTab.js

```diff
@@ -0,0 +1,200 @@
+import React, { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { Skeleton, Label } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { TableText, TableVariant, Thead, Tbody, Tr, Th, Td } from '@patternfly/react-table';
+import {
+  LongArrowAltUpIcon,
+  CheckIcon,
+} from '@patternfly/react-icons';
+import { selectAPIResponse } from 'foremanReact/redux/API/APISelectors';
+import { MODULE_STREAMS_KEY } from './ModuleStreamsConstants';
+import { selectModuleStreamStatus } from './ModuleStreamsSelectors';
+import { useBulkSelect } from '../../../../Table/TableHooks';
+import { getHostModuleStreams } from './ModuleStreamsActions';
+import InactiveText from '../../../../../scenes/ContentViews/components/InactiveText';
+import TableWrapper from '../../../../../components/Table/TableWrapper';
+
+export const ModuleStreamsTab = () => {
+  const hostDetails = useSelector(state => selectAPIResponse(state, 'HOST_DETAILS'));
+  const {
+    id: hostId,
+    name: hostName,
+  } = hostDetails;
+
+  const emptyContentTitle = __('This host does not have any Module streams.');
+  const emptyContentBody = __('Module streams will appear here when available.');
+  const emptySearchTitle = __('Your search returned no matching Module streams.');
+  const emptySearchBody = __('Try changing your search criteria.');
+
+  const columnHeaders = [
+    __('Name'),
+    __('State'),
+    __('Stream'),
+    __('Installation status'),
+    __('Installed profile'),
+  ];
+
+  const fetchItems = useCallback(
+    (params) => {
+      if (!hostId) return null;
```

Comment:
This function is a Redux action creator so it should always return a plain object and never `null`. You can return `hostIdNotReady` instead (see other tabs for examples-- you may need to rebase) :)